tests/net/conn_mgr_conn/src/main.c

```diff
@@ -0,0 +1,864 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/linker/sections.h>
+
+#include <zephyr/ztest.h>
+
+#include <zephyr/net/dummy.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/conn_mgr_connectivity.h>
+#include "test_conn_impl.h"
+
+/* This is a duplicate of conn_mgr_if_get_binding in net_if.c,
+ * which is currently not exposed.
+ */
```

Comment:
It is interesting that the test case needs to duplicate the function because it isn't exposed. Should we consider exposing it?

Reply:
I agree, I think it can and probably should be exposed, though I would like to do so in the next PR here:

https://github.com/zephyrproject-rtos/zephyr/pull/56549


The reason being, that PR includes a commit which reorganizes the file structure of `conn_mgr` and introduces `conn_mgr_private.h`, which would be an ideal place for `conn_mgr_if_get_binding` to live.

Reply:
That PR is not yet complete, but does now contain a commit doing as you suggest:

https://github.com/zephyrproject-rtos/zephyr/pull/56549/commits/08831405a3d856cbe46c8a55ce4e5fe411e899c6